lib/calculation_tasks.py

```diff
@@ -0,0 +1,93 @@
+    db_session = FLASK_DB.session
+
+    return _update_calculation_technique(db_session, calculation_technique_id, target_param_list)
+
+
+def _update_calculation_technique(db_session, calculation_technique_id, target_param_list):
+    calculation_technique = db_session.get(CalculationTechnique, calculation_technique_id)
+    calculation_technique.state = 'in_progress'
+    db_session.commit()
+
+    with tempfile.TemporaryDirectory() as tmp_dir_name:
+        try:
+            calculation_technique.calculations = []
+
+            for target_params in target_param_list:
+                bioreplicate_uuid        = target_params['bioreplicate_uuid']
+                subject_id               = target_params['subject_id']
+                subject_type             = target_params['subject_type']
+                measurement_technique_id = target_params['measurement_technique_id']
+
+                measurement_technique = db_session.get_one(
+                    MeasurementTechnique,
+                    measurement_technique_id,
+                )
+
+                calculation = Calculation(
+                    type=calculation_technique.type,
+                    subjectId=subject_id,
+                    subjectType=subject_type,
+                    measurementTechniqueId=measurement_technique_id,
+                    calculationTechniqueId=calculation_technique.id,
+                    bioreplicateUniqueId=bioreplicate_uuid,
+                )
+                db_session.add(calculation)
+                calculation_technique.calculations.append(calculation)
+
+                data = measurement_technique.get_subject_df(
+                    db_session,
+                    bioreplicate_uuid,
+                    subject_id,
+                    subject_type,
+                )
+
+                # If there is no standard deviation, remove it:
+                if data['std'].isna().all():
+                    data = data.drop(columns=['std'])
+
+                # Remove rows with NA values
+                data = data.dropna()
+
+                rscript = RScript(root_path=tmp_dir_name)
+                rscript.write_csv('input.csv', data)
+
+                script_name = f"scripts/modeling/{calculation_technique.type}.R"
+                output = rscript.run(script_name, 'input.csv', 'coefficients.json')
+
+                calculation.coefficients = rscript.read_json('coefficients.json')
+
+                calculation.state = 'ready'
+                calculation.calculatedAt = datetime.now(UTC)
+
+            calculation_technique.state = 'ready'
+            calculation_technique.error = None
+            db_session.commit()
+        except Exception as e:
+            db_session.rollback()
+            calculation_technique.state = 'error'
+
+            # TODO (2025-04-28) Make "error" a text field, show it somewhere
+            calculation_technique.error = str(e)[0:100]
+            db_session.commit()
+
+            raise
```

Comment:
I need to improve error handling a bit in general, I also should have a log of the operation of the R script and show it somewhere.